domain/agentbinary/state/state_test.go

```diff
@@ -263,3 +264,48 @@ func (s *stateSuite) TestGetObjectUUIDFailedObjectNotFound(c *gc.C) {
 	_, err := s.state.GetObjectUUID(context.Background(), "non-existent-path")
 	c.Check(err, jc.ErrorIs, agentbinaryerrors.ObjectNotFound)
 }
+
+func getMetadata(c *gc.C, db *sql.DB, objStoreUUID objectstore.UUID) agentbinary.Metadata {
```

Comment:
nit: Hi should this func be a stateSuite method? Esp since state_test is in the state package, and we should not be taking gc.C param in state.go

Reply:
The test file won't be built into the binary.
I don't see an issue here.